continuous-integration/readme.md

```diff
@@ -1,3 +1,235 @@
+
+We encourage engineering teams to make an upfront investment during Sprint 0 of a project to establish an automated and repeatable pipeline which continuously integrates code and releases system executable(s) to target cloud environments. Each integration should be verified by an automated build process that asserts a suite of validation tests pass and surface any errors across the developer team.
+
+We encourage teams to implement the CI/CD pipelines before any service code is written for customers, which usually happens in Sprint 0(N). This way, the engineering team can develop and test their work in isolation without impacting other developers and promote a consistent devops workflow throughout the engagement.
+
+These [principles](https://martinfowler.com/articles/continuousIntegration.html) map directly agile software development lifecycle [practices](https://en.wikipedia.org/wiki/Agile_software_development).
+
+<!-- markdownlint-disable MD036 -->
+**Table of contents**
+<!-- markdownlint-enable MD036 -->
+
+- [Continuous Integration](#continuous-integration)
+  - [Goals](#goals)
+  - [S/ingle Source Repository](#single-source-repository)
+  - [Build Automation](#build-automation)
+  - [Build Environment Dependencies](#build-environment-dependencies)
+  - [Infrastructure as Code](#infrastructure-as-code)
+    - [Sample DevOPS Workflow using Terraform and Cobalt](#sample-devops-workflow-using-terraform-and-cobalt)
+    - [Why](#why)
+    - [IAC DevOPS: Operations by Pull Request](#iac-devops-operations-by-pull-request)
+    - [Infrastructure Advocated Patterns](#infrastructure-advocated-patterns)
+    - [IAC Principles](#iac-principles)
+  - [Integration Validation](#integration-validation)
+  - [Git Driven Workflow](#git-driven-workflow)
+  - [Deliver Quickly and Daily](#deliver-quickly-and-daily)
+  - [Isolated Environments](#isolated-environments)
+  - [Developer Access to Latest Release Artifacts](#developer-access-to-latest-release-artifacts)
+  - [Integration Observability](#integration-observability)
+  - [Resources](#resources)
+
+## Goals
+
+Continuous integration automation is an integral part of the software development lifecycle intended to reduce build integration errors and maximize velocity across a dev crew.
+
+A robust build automation pipeline will:
+
+- Accelerate team velocity
+- Prevent integration problems
+- Avoid last minute chaos during release dates
+- Provide a quick feedback cycle for system-wide impact of local changes
+- Separate build and deployment stages
+- Measure and report metrics around build failures / success(s)
+- Increase visibility across the team enabling tighter communication
+- Reduce human errors, which is probably the most important part of automating the builds
+
+## S/ingle Source Repository
+
+- [ ] **Code / manifest artifacts required to build your project should be maintained in a single git repository.**
```

Comment:
I don't agree with this as an absolute point. See some commentary in the discussion for this PR - https://github.com/microsoft/code-with-engineering-playbook/pull/248 - which suggests something similar.

Proposing *minimizing* the amount of repositories is a good best practice, but enforcing *one* repository is an unrealistic proposition IMO as it can be driven by things that are out of our control (example, the customer already has some code we are working on).

I think the real problem is having a lot of repositories without a way to effectively manage build artifacts and versioned dependencies.